### PR TITLE
Rename UID to RequestId

### PR DESCRIPTION
**Pull request by notFloran**
#3

### CODE REVIEW

File: src/EventListener/RequestIdResponseListener.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+{
+    /**
+     * @var RequestIdProviderInterface
+     */
+    protected $requestIdProvider;
+
+    /**
+     * @param RequestIdProviderInterface $uidProvider
+     */
+    public function __construct(RequestIdProviderInterface $uidProvider)
```
Comment (nclsHart) [inline src/EventListener/RequestIdResponseListener.php]: $requestIdProvider ?

File: src/EventListener/RequestIdResponseListener.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+use Hexanet\Common\MonologExtraBundle\Provider\RequestId\RequestIdProviderInterface;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+
+class RequestIdResponseListener
+{
+    /**
+     * @var RequestIdProviderInterface
+     */
+    protected $requestIdProvider;
```
Comment (nclsHart) [inline src/EventListener/RequestIdResponseListener.php:13]: private

File: src/EventListener/RequestIdResponseListener.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    /**
+     * @param RequestIdProviderInterface $uidProvider
```
Comment (nclsHart) [inline src/EventListener/RequestIdResponseListener.php]: This comment should be omitted.

File: src/EventListener/RequestIdResponseListener.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    /**
+     * @param FilterResponseEvent $event
+     */
+    public function onKernelResponse(FilterResponseEvent $event) : void
+    {
+        $event->getResponse()->headers->set('X-Request-ID', $this->requestIdProvider->getRequestId());
+    }
+
```
Comment (nclsHart) [inline src/EventListener/RequestIdResponseListener.php]: extra space here

File: src/EventListener/RequestIdResponseListener.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+}
```
Comment (nclsHart) [inline src/EventListener/RequestIdResponseListener.php]: missing new line at eof.

File: src/EventListener/RequestIdResponseListener.php [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * @param RequestIdProviderInterface $uidProvider
+     */
+    public function __construct(RequestIdProviderInterface $uidProvider)
+    {
+        $this->requestIdProvider = $uidProvider;
+    }
+
+    /**
+     * @param FilterResponseEvent $event
```
Comment (nclsHart) [inline src/EventListener/RequestIdResponseListener.php]: This comment should be omitted too.

File: src/Processor/RequestIdProcessor.php [hunk 7]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Hexanet\Common\MonologExtraBundle\Processor;
+
+use Hexanet\Common\MonologExtraBundle\Provider\RequestId\RequestIdProviderInterface;
+
+class RequestIdProcessor
```
Comment (nclsHart) [inline src/Processor/RequestIdProcessor.php:7]: Most comments on `RequestIdResponseListener` are applicable here too.

File: src/Provider/RequestId/RequestIdProviderInterface.php [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+namespace Hexanet\Common\MonologExtraBundle\Provider\RequestId;
+
+interface RequestIdProviderInterface
+{
+    /**
+     * @return string
+     */
+    public function getRequestId() : string;
+}
```
Comment (nclsHart) [inline src/Provider/RequestId/RequestIdProviderInterface.php]: missing new line at eof

File: src/Provider/RequestId/RequestIdProviderInterface.php [hunk 9]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Hexanet\Common\MonologExtraBundle\Provider\RequestId;
+
+interface RequestIdProviderInterface
+{
+    /**
+     * @return string
```
Comment (nclsHart) [inline src/Provider/RequestId/RequestIdProviderInterface.php]: not needed phpdoc

File: src/Provider/RequestId/UniqidProvider.php [hunk 10]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Hexanet\Common\MonologExtraBundle\Provider\RequestId;
+
+class UniqidProvider implements RequestIdProviderInterface
+{
+    /**
+     * @var string
+     */
+    protected $requestId;
```
Comment (nclsHart) [inline src/Provider/RequestId/UniqidProvider.php]: private

File: src/Provider/RequestId/ApacheUniqueIdProvider.php [hunk 11]
```diff
@@ -1,37 +1,37 @@
...
+namespace Hexanet\Common\MonologExtraBundle\Provider\RequestId;
 
-class ApacheUniqueIdProvider implements UidProviderInterface
+class ApacheUniqueIdProvider implements RequestIdProviderInterface
 {
     /**
      * @var string
      */
-    protected $uid;
+    protected $requestId;
```
Comment (nclsHart) [inline src/Provider/RequestId/ApacheUniqueIdProvider.php]: private

File: src/Provider/RequestId/ApacheUniqueIdProvider.php [hunk 12]
```diff
@@ -1,37 +1,37 @@
 <?php
 
-namespace Hexanet\Common\MonologExtraBundle\Provider\Uid;
+namespace Hexanet\Common\MonologExtraBundle\Provider\RequestId;
 
-class ApacheUniqueIdProvider implements UidProviderInterface
+class ApacheUniqueIdProvider implements RequestIdProviderInterface
```
Comment (nclsHart) [inline src/Provider/RequestId/ApacheUniqueIdProvider.php]: IMO this class should be marked as final.

File: src/Provider/RequestId/UniqidProvider.php [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+
+    public function __construct()
+    {
+        $this->requestId = uniqid();
+    }
+
+    /**
+     * @return string
```
Comment (nclsHart) [inline src/Provider/RequestId/UniqidProvider.php]: not needed phpdoc

**Review by nclsHart — CHANGES_REQUESTED** [review]

File: src/EventListener/RequestIdResponseListener.php (hunk 4 above)
Comment (notFloran) [reply src/EventListener/RequestIdResponseListener.php]: theo style

**Review by nclsHart — APPROVED** [review]

**Pull request merged**